src/screen_capture/ScreenCapture.cpp

```diff
@@ -0,0 +1,226 @@
+
+        // GetDIBits requires format info about the bitmap. We can have GetDIBits
+        // fill a structure with that info if we pass a NULL pointer for lpvBits:
+        // Reserve memory for bitmap info (BITMAPINFOHEADER + largest possible
+        // palette):
+        if ((lpbi = (LPBITMAPINFO) (new char[sizeof(BITMAPINFOHEADER) + 256 * sizeof(RGBQUAD)])) == NULL)
+        {
+            return false;
+        }
+
+
+        ZeroMemory(&lpbi->bmiHeader, sizeof(BITMAPINFOHEADER));
+        lpbi->bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+        // Get info but first de-select OffscrBmp because GetDIBits requires it:
+        SelectObject(OffscrDC, OldBmp);
+        if (!GetDIBits(OffscrDC, OffscrBmp, 0, height, NULL, lpbi, DIB_RGB_COLORS))
+        {
+            return false;
+        }
+
+        // Reserve memory for bitmap bits:
+        if ((lpvBits = new char[lpbi->bmiHeader.biSizeImage]) == NULL)
+        {
+            return false;
+        }
+
+        // Have GetDIBits convert OffscrBmp to a DIB (device-independent bitmap):
+        if (!GetDIBits(OffscrDC, OffscrBmp, 0, height, lpvBits, lpbi, DIB_RGB_COLORS))
+        {
+            return false;
+        }
+
+        // Create a file to save the DIB to:
+        if ((BmpFile = CreateFile(filename,
+                                  GENERIC_WRITE,
+                                  0, NULL,
+                                  CREATE_ALWAYS,
+                                  FILE_ATTRIBUTE_NORMAL,
+                                  NULL)) == INVALID_HANDLE_VALUE)
+        {
+            return false;
+        }
+
+        DWORD Written;    // number of bytes written by WriteFile
+
+        // Write a file header to the file:
+        bmfh.bfType = 19778;        // 'BM'
+        // bmfh.bfSize = ???        // we'll write that later
+        bmfh.bfReserved1 = bmfh.bfReserved2 = 0;
+        // bmfh.bfOffBits = ???     // we'll write that later
+        if (!WriteFile(BmpFile, &bmfh, sizeof(bmfh), &Written, NULL))
+        {
+            return false;
+        }
+
+        if (Written < sizeof(bmfh))
+        {
+            return false;
+        }
+
+        // Write BITMAPINFOHEADER to the file:
+        if (!WriteFile(BmpFile, &lpbi->bmiHeader, sizeof(BITMAPINFOHEADER), &Written, NULL))
+        {
+            return false;
+        }
+
+        if (Written < sizeof(BITMAPINFOHEADER))
+        {
+            return false;
+        }
+
+        // Calculate size of palette:
+        int PalEntries;
+        // 16-bit or 32-bit bitmaps require bit masks:
+        if (lpbi->bmiHeader.biCompression == BI_BITFIELDS)
+        {
+            PalEntries = 3;
+        }
+        else
+        {
+            // bitmap is palettized?
+            PalEntries = (lpbi->bmiHeader.biBitCount <= 8) ?
+                         // 2^biBitCount palette entries max.:
+                         (int) (1 << lpbi->bmiHeader.biBitCount)

```

Comment:
I guess you copy past a C function from internet, or maybe you must use this library like this , i don't know, but if it's not the case, change your code to use reference, static_cast and so on